src/WMOWithRequest/WMOWithRequest.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+import DefaultLoading from './DefaultLoading';
+import { customRetryError } from './DefaultRetryError';
+
+export const WMORequestErrors = {
+  CLIENT_ERROR: 'CLIENT_ERROR',
+  SERVER_ERROR: 'SERVER_ERROR',
+  NETWORK_ERROR: 'NETWORK_ERROR'
+};
+
+function withRequestWrapper(
+  WrappedComponent,
+  {
+    onRequest,
+    LoadingComponent = DefaultLoading,
+    loading,
+    error,
+    NetworkErrorComponent = customRetryError('Network error'),
+    ServerErrorComponent = customRetryError('Server error')
+  }
+) {
+  class WithRequestComponent extends Component {
+    componentWillMount() {
+      this.handleRequest();
+    }
+
+    handleRequest = () => {
+      onRequest(this.props);
+    };
+
+    render() {
+      if (loading(this.props)) {
+        return <LoadingComponent />;
+      }
+
+      const requestError = error(this.props);
+
+      if (requestError === WMORequestErrors.NETWORK_ERROR) {
+        return <NetworkErrorComponent onRetry={this.handleRequest} />;
```

Comment:
Maybe we can reduce `render`'s complexity eliminating these `if`s:
```
// Object that contains, according to requestError, different ErrorComponents.
// This will require some refactoring, especially to write `WMORequestErrors.NETWORK_ERROR` as a field
 
const responses = { WMORequestErrors.NETWORK_ERROR: NetworkErrorComponent ... };
const ErrorComponent = responses[requestError];

return <ErrorComponent onRetry={this.handleRequest} />;
```